privval/file.go

```diff
@@ -402,6 +403,13 @@ func (pv *FilePV) signVote(chainID string, vote *tmproto.Vote) error {
 		return err
 	}
 	vote.Signature = sig
+
```

Comment:
After going through this part of the code, my first impression was that `ExtensionSignature` should receive the same treatment as a `Signature` (so it should be part of `lss`, and should be set next to line 392.
But then, I read @williambanfield's comment above and you have a point in that, at least in the current spec, a validator signing two different extensions for the same vote is not considered equivocation. So I tend to agree with William's proposal here: the extension need to be set next to line 392, but we don't need to modify the `lss`.

I'm just thinking whether it would make sense to replace field `Round` in `CanonicalVoteExtension` by `blockID`, as all we want to do is link an extension to a block, regardless of round. Anyway, this is just a minor internal detail, as we can currently derive the block ID from the vote we are extending, provided height and round both match in the canonical vote and the canonical extension.

Reply:
I've updated the logic here - please check for me that it makes sense to you too?

Reply:
It does to me :-)